packages/http/test/index.js

```diff
@@ -509,13 +509,54 @@ describe('post', () => {
       return body;
     });
 
+    testServer
+      .post('/api/csv-reader')
+      .times(3)
+      .reply(200, function (url, body) {
+        console.log('thanks for the message!', body);
+        return body;
+      });
+
     testServer
       .post('/api/fake-custom-success-codes')
       .reply(302, function (url, body) {
         return { ...body, statusCode: 302 };
       });
   });
 
+  it('should make an http request from inside the parseCSV callback', async function () {
+    const csv = 'id,name\n1,taylor\n2,mtuchi\n3,joe\n4,stu\n5,elias';
+    const state = { references: [], data: [], apiResponses: [] };
+
+    const resultingState = await parseCsv(
+      csv,
+      { chunkSize: 2 },
+      async (state, rows) => {
+        // TODO: how do we write these in the real world? Like this?
+        if (rows.length > 0)
+          await post(
+            'https://www.example.com/api/csv-reader',
+            {
+              body: rows,
+            },
+            state => {
+              console.log("I'm a regular callback");
```

Comment:
@taylordowns2000 Coming in out of context so I may be missing something, but isn't the http response now on state.data here?

So you can just `state.apiReferences.push(state.data)`

If we're not seeing that at runtime, that feels like a bug